aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -160,11 +137,65 @@ private void recurseTree(
         }
     }
 
+    private Set<String> childrenOf(
+            QueryField parentIdField,
+            Collection<String> parentIds,
+            ModelSchema childSchema
+    ) {
+        SQLiteTable childTable = SQLiteTable.fromSchema(childSchema);
+        String childPrimaryKey = childTable.getPrimaryKey().getAliasedName();
+
+        // Chain predicates with OR operator.
+        // Beware the SQLite expression tree size limit of 1000
+        List<QueryOptions> options = new ArrayList<>();
+        QueryPredicate predicate = QueryPredicates.none();
+        int currentSize = 0;
+        for (String parentId : parentIds) {
+            // Stop appending if size exceeds 900
+            if (currentSize > 900) {
+                options.add(Where.matches(predicate));
+                predicate = QueryPredicates.none();
+                currentSize = 0;
+            }
+            QueryPredicate operation = parentIdField.eq(parentId);
+            predicate = predicate.or(operation);
+            currentSize++;
+        }
+        options.add(Where.matches(predicate));
+
+        // Collect every children one level deeper than current level
+        // SELECT * FROM <CHILD_TABLE> WHERE <PARENT> = <ID_1> OR <PARENT> = <ID_2> OR ...
```

Comment:
Instead of chaining with SQL `OR` keyword, could you build the query like this?
```suggestion
// SELECT * FROM <CHILD_TABLE> WHERE <PARENT> IN (<ID_1>, <ID_2>, ...)
```
I think there are two benefits to this approach:
1. It would prevent the SQLite expression tree size limit of 1000, without having to break things up into multiple queries
2. It would probably improve performance.  I don't know for sure, but I would guess that SQLite represents the list as a hash table internally.  If that is true, then this approach would improve the overall efficiency of this query from `O(n^2)` to `O(n)`.

Reply:
Turns out [SQLite optimizer](https://www.sqlite.org/optoverview.html) converts `column = a OR column = b OR column = c` automatically into `column IN (a, b, c)`. Cool!

This approach solved the tree depth limit problem since it is a single expression rather than a combination of 1000+. However, it turns out that Android SQLit _also_ has a hard limit on the number of `?` that you can use per query expression (same as tree depth limit of 1000). 

I resolved it by simply using ID strings directly without binding them, which could potentially be a problem.
As of right now, ID cannot be anything than UUID format, so there is no concern for SQL injection. However, there is a conversation floating around in favor of allowing _any_ string to be assignable to ID field (like iOS does), so I'm curious if this change would potentially introduce a vulnerability.